sway/sway_text_node.c

```diff
@@ -163,6 +164,39 @@ static void render_backing_buffer(struct text_buffer *buffer) {
 	cairo_font_options_destroy(fo);
 }
 
+static void text_calc_size(struct text_buffer *buffer) {
+	struct sway_text_node *props = &buffer->props;
+	props->width = 0;
+
+	cairo_t *c = cairo_create(NULL);
+	if (!c) {
+		sway_log(SWAY_ERROR, "cairo_t allocation failed");
+		return;
+	}
+
+	cairo_set_antialias(c, CAIRO_ANTIALIAS_BEST);
+	for (int i = 0; i < root->outputs->length; ++i) {
+		struct sway_output *output = root->outputs->items[i];
+		if (!output->wlr_output->enabled) {
+			continue;
+		}
+
+		int size, baseline;
+		get_text_size(c, config->font_description, &size, NULL,
+			&baseline, output->wlr_output->scale, props->pango_markup, "%s", buffer->text);
```

Comment:
I wonder if it would be fine to just take the largest scale from all outputs.

`get_text_size` is quite expensive - a quick test on my Ryzen laptop on battery power clocks `get_text_size` with a random title around 400µs, although it gets a lot faster on external power of course.